apis/postgresql/v1alpha1/grant_types.go

```diff
@@ -0,0 +1,186 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/reference"
+)
+
+// A GrantSpec defines the desired state of a Grant.
+type GrantSpec struct {
+	xpv1.ResourceSpec `json:",inline"`
+	ForProvider       GrantParameters `json:"forProvider"`
+}
+
+// GrantPrivilege represents a privilege to be granted
+// +kubebuilder:validation:Pattern:=[A-Z]+
+type GrantPrivilege string
+
+// If Privileges are specified, we should have at least one
+
+// GrantPrivileges is a list of the privileges to be granted
+// +kubebuilder:validation:MinItems:=1
+type GrantPrivileges []GrantPrivilege
+
+// ToStringSlice converts the slice of privileges to strings
+func (gp *GrantPrivileges) ToStringSlice() []string {
```

Comment:
I've added a nil pointer check returning an empty string slice just in case, although in practice I wasn't able to trigger a nil pointer deref as the privileges field is typechecked both at the API (if you specify the field, it must be a slice with at least one item) and during sanity checking before being used (returning `errNoPrivileges`).